### PR TITLE
feat(staking): [LW-7508] add un-delegation functionality

### PR DESCRIPTION
**Pull request by xdzurman**
# Checklist

- [x] JIRA - \<link>
- [ ] Proper tests implemented
- [x] Screenshots added.

---

## Proposed solution

Always allow removing pools from preferences drawer. If the last pool was removed, render the sad smiley content. I chose the easier alternative to get this out quick - i.e., you can remove pools not just in overview Management drawer, but also in Browse pools to not complicate the code. Also adjusted the result screen (see screenshots).

## Testing

Manage current portfolio > Remove all pools > continue, sign the transaction

## Screenshots

image
image
image
image

---
# Allure report
`allure-report-publisher` generated test report!

**smokeTests**: ✅ [test report](https://lace-qa:8PP5wBaDV6UoXj@dq4ajm5i5q7bz.cloudfront.net/linux/chrome/2820/6742247635/index.html) for [430d524c](https://github.com/input-output-hk/lace/pull/634/commits/430d524c42d537b8f4150ea4678b6b4d1a664f0c)
|       | passed | failed | skipped | flaky | total | result |
|-------|--------|--------|---------|-------|-------|--------|
| Total | 31     | 0      | 0       | 0     | 31    | ✅     |

### CODE REVIEW

File: packages/staking/src/features/drawer/StakePoolDetails.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ type DraftPortfolioInvalidReason = 'invalid-allocation' | 'slider-zero';
 type DraftPortfolioValidity = { valid: true } | { valid: false; reason: DraftPortfolioInvalidReason };
 
 const getDraftPortfolioValidity = (store: DelegationPortfolioStore): DraftPortfolioValidity => {
-  if (!store.draftPortfolio || store.draftPortfolio.length === 0) return { valid: true }; // throw new Error('Draft portfolio is not defined');
+  if (!store.draftPortfolio || store.draftPortfolio.length === 0) return { valid: true };
```
Comment (refi93) [inline packages/staking/src/features/drawer/StakePoolDetails.tsx:37]: nitpick - I'd consider re-writing it a bit more concisely:

```suggestion
  if (!store.draftPortfolio?.length) return { valid: true };
```

File: packages/staking/src/features/drawer/preferences/empty.svg
Comment (refi93) [inline packages/staking/src/features/drawer/preferences/empty.svg:1]: I'd consider giving this a more descriptive name, perhaps `sad-smiley.svg`. Moreover, took me a while to realize that `<Empty />` actually refers to a pricture and not some placeholder "empty" component, `<SadSmiley />` would be obvious at first sight

File: packages/staking/src/features/drawer/preferences/NoPoolsSelected.tsx [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  return (
+    <div className={styles.container}>
+      <Empty className={styles.icon} />
+      <Text.Body.Normal className={styles.text} weight="$semibold">
+        {t('drawer.preferences.noSelectedPools')}
+      </Text.Body.Normal>
+      <div className={styles.button}>
+        <Button.CallToAction label={t('drawer.preferences.browsePools')} onClick={onBrowsePoolsButtonClick} />
+      </div>
+    </div>
```
Comment (refi93) [inline packages/staking/src/features/drawer/preferences/NoPoolsSelected.tsx:21]: nitpick - I'd consider using the ui component `Box` which leverages the `@lace/ui` framework and I'd consider dropping the button style as it seems overall the preferred way in the codebase to use the `mt` prop. Moreover, I'd consider moving the "weight" property of the `Text.Body.Normal` within the styles to keep the styles for the respective component in a single place

```suggestion
    <Box className={styles.container}>
      <Empty className={styles.icon} />
      <Text.Body.Normal className={styles.text}>
        {t('drawer.preferences.noSelectedPools')}
      </Text.Body.Normal>
      <Box mt="$40">
        <Button.CallToAction label={t('drawer.preferences.browsePools')} onClick={onBrowsePoolsButtonClick} />
      </Box>
    </Box>
```

**Review by refi93 — APPROVED** [review]
only minor suggestions, other than that LGTM

File: packages/staking/src/features/drawer/StakePoolDetails.tsx (hunk 1 above)
Comment (szymonmaslowski) [reply packages/staking/src/features/drawer/StakePoolDetails.tsx:37]: I think our lint rules will allow for
```suggestion
  if (store.draftPortfolio?.length !== 0) return { valid: true };
```

File: packages/staking/src/features/drawer/preferences/NoPoolsSelected.css.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+export const icon = style({
+  fontSize: theme.spacing.$112,
+});
+
+export const text = style({
+  color: theme.colors.$greyText,
+});
+
+export const button = style({
```
Comment (szymonmaslowski) [inline packages/staking/src/features/drawer/preferences/NoPoolsSelected.css.ts:18]: it is really a nitpick, but I don't have other things to comment so... 😆 this class might be removed if you add margin-bottom to the text class 😉

**Review by szymonmaslowski — APPROVED** [review]
Just a few minor nitpicks. Approving already 😉

File: packages/staking/src/features/Drawer/preferences/sad-smiley.svg
Comment (przemyslaw-wlodek) [inline packages/staking/src/features/Drawer/preferences/sad-smiley.svg:1]: Let's discuss placing assets in the common dir in staking, like @refi93 [suggested](https://github.com/input-output-hk/lace/pull/659/files#r1369308748).

**Review by przemyslaw-wlodek — CHANGES_REQUESTED** [review]

**Review by przemyslaw-wlodek — APPROVED** [review]
Sorry, I wanted to approve 😄 These are just a couple of minor improvements that can be done.

**Comment by xdzurman** [comment]
Blocked by a bug in js sdk - https://input-output.atlassian.net/browse/LW-8993

**Review by ljagiela — CHANGES_REQUESTED** [review]
My issues described in Jira.

**Pull request merged**